File: src/main/java/com/tukan/api/entity/Assessment.java
package com.tukan.api.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "assessment", indexes = {
        @Index(name = "idx_assessment_user_id", columnList = "user_id", unique = true)
})
@Getter
@Setter
@NoArgsConstructor
public class Assessment {

    public enum NutritionalGoal {
        WEIGHT_LOSS,
        MUSCLE_GAIN,
        MAINTENANCE,
        DIETARY_REEDUCATION,
        SPORTS_PERFORMANCE
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(name = "goal", nullable = false)
    private NutritionalGoal goal;

    @Column(name = "dietary_restrictions", length = 500)
    private String dietaryRestrictions;

    @Column(name = "allergies", length = 500)
    private String allergies;

    @Column(name = "health_conditions", length = 500)
    private String healthConditions;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }
}

File: frontend/src/types/assessment.ts
import type { Dashboard, DietType } from './api'

export interface CreateAssessmentRequest {
  goal: Dashboard.NutritionalGoal
  dietType?: DietType
  dietaryRestrictions?: string
  allergies?: string
  healthConditions?: string
  mealsPerDay?: number
  targetWeightKg?: number
}

export interface UpdateAssessmentRequest {
  goal?: Dashboard.NutritionalGoal
  dietType?: DietType
  dietaryRestrictions?: string
  allergies?: string
  healthConditions?: string
  mealsPerDay?: number
  targetWeightKg?: number
}

